staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types_discovery.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIGroupDiscoveryList is a resource containing a list of APIGroupDiscoveries.
+// The shape is chosen to be very similar to "normal" api endpoints and work well with our type system
```

Comment:
This is more of a code comment than an API comment.  Instead, this should be in the API form - assume you're communicating to an end user who sees this object type show up in their JSON response - what do they need to know?  Usually it would be (rough order):

1. What does the object's purpose in most common use (where a user sees it)? Contains a list of API groups
2. What is additional information the user needs to know how to use it?
3. Is there any ambiguity in the fields of the object that is better explained here vs in multiple field docs?

We do a generally poor job of this, but always start from the perspective that this comment is the singular place a user goes to understand the object showing up in a response.  Generated API docs and `kubectl explain` will also display it prominently.  2-3 sentences is usually minimum, try to concisely describe how the user uses this object.